R/render.R

```diff
@@ -604,11 +604,13 @@ render <- function(input,
       assign("params", params, envir = envir)
       lockBinding("params", envir)
       on.exit({
-        do.call("unlockBinding", list("params", envir))
-        if (hasParams)
-          assign("params", envirParams, envir = envir)
-        else
-          remove("params", envir = envir)
+        if ('params' %in% ls(envir = envir)) {
```

Comment:
A simpler check would be `exists("params", envir = envir, inherits = FALSE)` (avoids listing all objects in associated environment)